File: Eyes/ImageTester/src/csharp/lib/TestObjects/Batch.cs
using System;
using System.Collections.Generic;
using System.IO;
using Applitools.Images;

namespace Applitools.ImageTester.TestObjects
{
    public class Batch : TestUnit
    {
        private BatchInfo batch;
        private readonly IList<Test> tests = new List<Test>();

        public Batch(FileStream file) : base(file)
        {
        }

        public Batch(BatchInfo batch) : base(batch.Name)
        {
            this.batch = batch;
        }

        public override void Run(Eyes eyes)
        {
            batch = batch == null ? new BatchInfo(Name()) : batch;
            eyes.Batch = batch;
            Console.WriteLine("Batch: {0}", Name());
            foreach (var test in tests)
            {
                try
                {
                    test.Run(eyes);
                }
                finally
                {
                    test.Dispose();
                }
            }

            eyes.Batch = null;
        }

        public virtual void AddTest(Test test)
        {
            tests.Add(test);
        }

        public override void Dispose()
        {
            if (tests == null)
            {
                return;
            }

            foreach (var test in tests)
            {
                test.Dispose();
            }
        }
    }
}

File: Eyes/ImageTester/src/lib/java/com/applitools/ImageTester/TestObjects/Batch.java
package com.applitools.ImageTester.TestObjects;

import com.applitools.eyes.BatchInfo;
import com.applitools.eyes.images.Eyes;

import java.io.File;
import java.util.LinkedList;
import java.util.Queue;

public class Batch extends TestUnit {
    private BatchInfo batch_;
    private Queue<Test> tests_ = new LinkedList<Test>();

    public Batch(File file) {
        super(file);
    }

    public Batch(BatchInfo batch) {
        super(batch.getName());
        batch_ = batch;
    }

    public void run(Eyes eyes) {
        batch_ = batch_ == null ? new BatchInfo(name()) : batch_;
        eyes.setBatch(batch_);
        System.out.printf("Batch: %s\n", name());
        for (Test test : tests_) {
            try {
                test.run(eyes);
            } finally {
                test.dispose();
            }
        }
        eyes.setBatch(null);
    }

    public void addTest(Test test) {
        tests_.add(test);
    }

    public void dispose() {
        if (tests_ == null) return;
        for (Test test : tests_) {
            test.dispose();
        }
    }
}
